sensing/imu.cpp

```diff
@@ -356,49 +344,51 @@ bool Imu::stop_magnetic_north_calibration(void)
 }
 
 
-const bool Imu::is_ready(void) const
+bool Imu::is_ready(void) const
 {
-    return is_ready_;
+    return !is_calibration_ongoing();
 }
 
 
-void Imu::do_startup_calibration(void)
+bool Imu::is_calibration_ongoing(void) const
 {
-    if (is_ready_ == false)
-    {
-        // Make sure calibration is ongoing
-        if (do_gyroscope_bias_calibration_ == false)
-        {
-            start_gyroscope_bias_calibration();
-            timestamp_gyro_stable = time_keeper_get_s();
-        }
-
-        // Check if the gyroscope values are stable
-        bool gyro_is_stable = (maths_f_abs(config_.gyroscope.mean_values[X] - scaled_gyro_[X]) < config_.startup_calib_gyro_threshold)
-                              && (maths_f_abs(config_.gyroscope.mean_values[Y] - scaled_gyro_[Y]) < config_.startup_calib_gyro_threshold)
-                              && (maths_f_abs(config_.gyroscope.mean_values[Z] - scaled_gyro_[Z]) < config_.startup_calib_gyro_threshold);
-
-
-        if (gyro_is_stable == false)
-        {
-            is_ready_ = false;
-            // Reset timestamp
-            timestamp_gyro_stable = time_keeper_get_s();
```

Comment:
Should 'timestamp_gyro_stable' not indicate for how long the calibration is ongoing? Why is it updated if gyro is not stable yet? This might be the reason why startup calibration is taking to long.

Also, IMHO, it should be renamed to something like calibration_start_time